src/qt/macos_appnap.mm

```diff
@@ -0,0 +1,79 @@
+#include "macos_appnap.h"
+
+#include <AvailabilityMacros.h>
+#include <Foundation/NSProcessInfo.h>
+#include <Foundation/Foundation.h>
+
+#if defined(Q_OS_OSX) && (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_9)
+typedef quint64 NSActivityOptions;
+enum {
+    // To include one of these individual flags in one of the sets, use bitwise or:
+    //   NSActivityUserInitiated | NSActivityIdleDisplaySleepDisabled
+    // (this may be used during a presentation, for example)
+
+    // To exclude from one of the sets, use bitwise and with not:
+    //   NSActivityUserInitiated & ~NSActivitySuddenTerminationDisabled
+    // (this may be used during a user intiated action that may be safely terminated with no application interaction in case of logout)
+
+    // Used for activities that require the screen to stay powered on.
+    NSActivityIdleDisplaySleepDisabled = (1ULL << 40),
+
+    // Used for activities that require the computer to not idle sleep. This is included in NSActivityUserInitiated.
+    NSActivityIdleSystemSleepDisabled = (1ULL << 20),
+
+    // Prevents sudden termination. This is included in NSActivityUserInitiated.
+    NSActivitySuddenTerminationDisabled = (1ULL << 14),
+
+    // Prevents automatic termination. This is included in NSActivityUserInitiated.
+    NSActivityAutomaticTerminationDisabled = (1ULL << 15),
+
+    // ----
+    // Sets of options.
+
+    // App is performing a user-requested action.
+    NSActivityUserInitiated = (0x00FFFFFFULL | NSActivityIdleSystemSleepDisabled),
+    NSActivityUserInitiatedAllowingIdleSystemSleep = (NSActivityUserInitiated & ~NSActivityIdleSystemSleepDisabled),
+
+    // App has initiated some kind of work, but not as the direct result of user request.
+    NSActivityBackground = 0x000000FFULL,
+
+    // Used for activities that require the highest amount of timer and I/O precision available. Very few applications should need to use this constant.
+    NSActivityLatencyCritical = 0xFF00000000ULL,
+};
+#endif
+
+class CAppNapInhibitor::Private
+{
+public:
+    NSObject* activityId;
+};
+
+CAppNapInhibitor::CAppNapInhibitor(const char* strReason) : d(new Private)
+{
+    const NSActivityOptions activityOptions =
```

Comment:
Since this isn't defined for < MAC_OS_X_VERSION_10_9, does it need a runtime check?

Reply:
No idea, i can't setup 10.8 build vm to test code properly, so pre 10.9 support removed.